src/Factory/SwaggerSchemaFactory.php

```diff
@@ -73,12 +73,31 @@ protected function resolveSchemaFile($schemaFile)
                 );
         }
 
-        $refResolver = new RefResolver(
+        $schemaStorage = new SchemaStorage(
             $uriRetriever,
             new UriResolver()
         );
 
-        return $refResolver->resolve($schemaFile);
+        $schema = $schemaStorage->getSchema($schemaFile);
+
+        // JsonSchema normally defers resolution of $ref values until validation.
+        // That does not work for us, because we need to have the complete schema
+        // to build definitions.
+        $this->expandSchemaReferences($schema, $schemaStorage);
```

Comment:
This was annoying to track down but my YAML schema now works properly.

Reply:
Nicely stopped. The new public API of JSON Schema is confusing as Hell. I've experimented quite a bit the the library and I had some ideas to improve the implementation. But this solution will do nicely.